plugins/test/unit/plugins/importers/yum/repomd/test_metadata.py

```diff
@@ -205,6 +206,50 @@ def test_does_not_skip_unknown_sqlite_files(self):
         self.assertTrue(requests[1].destination.endswith('pkgtags.sqlite.gz'))
 
 
+class TestQueryAuthToken(unittest.TestCase):
+    def setUp(self):
+        self.qstring = '?foo'
+        self.url_modify = RepoURLModifier(query_auth_token=self.qstring[1:])
+        self.metadata_files = metadata.MetadataFiles('http://pulpproject.org',
+                                                     '/a/b/c',
+                                                     DownloaderConfig(),
+                                                     self.url_modify)
+
+    def test_repo_url(self):
+        self.assertTrue(self.metadata_files.repo_url.endswith(self.qstring))
+
+    # DownloadRequest is already in metadata, so it needs to be mocked in-module
+    @mock.patch('pulp_rpm.plugins.importers.yum.repomd.metadata.DownloadRequest', autospec=True)
+    def test_download_repomd(self, mock_download_request):
+        self.metadata_files.download_repomd()
+        try:
+            self.metadata_files.download_repomd()
+        except:
+            pass
```

Comment:
What Exception are you expecting here? Can you catch that exception explicitly and add a comment explaining why you expect an Exception on the second call?